fetch/origin/assorted.window.js

```diff
@@ -194,8 +195,16 @@ function referrerPolicyTestString(referrerPolicy, destination) {
     promise_test(fetchReferrerPolicy(testObj.policy,
                                      destination.name,
                                      "cors",
-                                     origins.HTTP_ORIGIN),
+                                     origins.HTTP_ORIGIN,
+                                     "POST"),
                  referrerPolicyTestString(testObj.policy,
-                                          destination.name + " fetch cors mode"));
+                                          destination.name + " fetch cors mode (POST)"));
+    promise_test(fetchReferrerPolicy(testObj.policy,
```

Comment:
mode = "cors", method = "GET", name = "same-origin" => I think no origin should be attached, but this expects origins.HTTP_ORIGIN.

Reply:
In that case https://github.com/whatwg/fetch/pull/1013 should check for both response tainting and mode, right?

Reply:
IIUC correctly yutakahirano's testcase is very similar to what got me started, except I was thinking about POST and thus having to deal with referrer-policy.

I think the current spec does have the no origin outcome. So if that is the behavior we want to go for then maybe all that is needed is to add the test and expectation for the GET CORS requests (and having to fix the implementations).

I guess some more feedback from yutakahirano about why the outcome should be no origin would be nice; I can fix the test depending on what we decide.

Reply:
Ah, I suspect checking mode and response tainting for "cors" is equivalent to only checking response tainting, indeed.

Are the other tests correct though? If you have a "cors" same-origin POST it would still be subject to referrer policy per https://fetch.spec.whatwg.org/#append-a-request-origin-header but the tests assert otherwise.

Reply:
Yes, if we stick to the current spec than the POST and GET "cors" requests should be in sync, i.e for the former take the referrer-policy into account and the latter expects no origin. I will update the test soon.

Reply:
Test was updated.